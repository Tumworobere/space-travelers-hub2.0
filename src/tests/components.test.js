import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Naver from '../Components/Navbar';
import Footer from '../Components/Footer';
import RocketContainer from '../Components/Rockets/RocketContainer';
import MissionsContainer from '../Components/missions/MissionsContainer';
import ProfileContainer from '../Components/profiles/ProfileContainer';

describe('Series of tests rendering of each major components', () => {
  it('rendering of Navigation bar(Header) correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter to="/" className="navbar-brand">
          <Naver />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('rendering footer correctly', () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Rendering Rockets page correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Rendering Missions page correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionsContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Rendering Profiles Page correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProfileContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
