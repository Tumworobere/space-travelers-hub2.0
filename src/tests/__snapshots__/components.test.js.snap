// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series of tests rendering of each major components Rendering Missions page correctly 1`] = `
<div
  className="container-fluid h-auto py-5"
>
  <div
    className="px-3 py-3"
  >
    <div
      className="row"
    >
      <div
        className="col-2 bords"
      >
        <p
          className="h3"
        >
          Mission
        </p>
      </div>
      <div
        className="col-6 bords"
      >
        <p
          className="h3"
        >
          Description
        </p>
      </div>
      <div
        className="col-2 bords"
      >
        <p
          className="h3"
        >
          Status
        </p>
      </div>
      <div
        className="col-2 bords"
      />
    </div>
  </div>
</div>
`;

exports[`Series of tests rendering of each major components Rendering Profiles Page correctly 1`] = `
<div
  className="container w-100 d-flex"
>
  <div
    className="w-50 me-3"
  >
    <h3
      className="h3"
    >
      My Missions
    </h3>
    <ul
      className="list-group"
    />
    <p
      className="p-3 lead"
    >
      No Missions Reserved Yet
    </p>
  </div>
  <div
    className="w-50 ms-3"
  >
    <h3
      className="h3"
    >
      My Rockets
    </h3>
    <ul
      className="list-group"
    />
    <p
      className="p-3 lead"
    >
      No Rockets Reserved Yet
    </p>
  </div>
</div>
`;

exports[`Series of tests rendering of each major components Rendering Rockets page correctly 1`] = `
<div
  className="container-fluid"
/>
`;

exports[`Series of tests rendering of each major components rendering footer correctly 1`] = `
<div
  className="footers"
>
  <footer
    className="bg-dark py-2"
  >
    <div
      className="container"
    >
      <p
        className="text-white text-center"
      >
        Copyright © 2022 designed by Mubarak A Ibrahim and Tumworobere Annah
      </p>
    </div>
  </footer>
</div>
`;

exports[`Series of tests rendering of each major components rendering of Navigation bar(Header) correctly 1`] = `
<div
  className="navs navigate"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-lignt lead"
  >
    <div
      className="container"
    >
      <a
        aria-current="page"
        className="navbar-brand fw-normal d-flex active"
        href="/"
        id="title"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="pe-3"
          id="logos-img"
          src="planet.png"
        />
        <h3>
          Space Travelers‘ Hub
        </h3>
      </a>
      <button
        className="navbar-toggler"
        data-bs-target="#navbar"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse justify-content-end"
      >
        <ul
          className="navbar-nav topnav"
        >
          <li
            className="nav-item pe-4"
          >
            <a
              aria-current="page"
              className="nav-link fw-normal active"
              href="/"
              onClick={[Function]}
            >
              Rockets
            </a>
          </li>
          <li
            className="nav-item pe-4 "
          >
            <a
              className="nav-link fw-normal"
              href="/missions"
              onClick={[Function]}
            >
              Missions
            </a>
          </li>
          <li
            className="list-child"
          />
          <li
            className="nav-item pe-4"
          >
            <a
              className="nav-link fw-normal"
              href="/my-profile"
              onClick={[Function]}
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
